### ISSUE TITLE
validation.undeclaredVariableCheck breaks in 5.x

### ISSUE THREAD

**Issue by zaggino** [babel/babel#1195]
I've following snippet of code:
```
function forwardCall(reply, path, method, payload = null) {
  let opts = {
    uri: constructCoreUrl(path),
    json: true,
    method: method,
    headers: config.get('headers'),
    resolveWithFullResponse: true
  };

  if (method === 'PUT' || method === 'POST' || method === 'PATCH') {
    opts.body = payload;
  }
```
with babel `4.7.16` and optional `validation.undeclaredVariableCheck` it works fine
with babel `5.0.10` **and without** optional `validation.undeclaredVariableCheck` it works fine
with babel `5.0.10` and optional `validation.undeclaredVariableCheck` the code breaks with
```
$ ReferenceError : routes.js: Line 30: Reference to undeclared variable "method"
```
Line 30 is the  `method: method,` line

**Comment by sebmck** [comment]
Can you try this on 5.0.11 please, I may have inadvertently fixed this also.

**Comment by zaggino** [comment]
Yes, seems fixed in `5.0.11`
